rero_ils/modules/collections/views.py

```diff
@@ -0,0 +1,137 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Blueprint used for loading templates."""
+
+from __future__ import absolute_import, print_function
+
+from flask import Blueprint, current_app, render_template
+from invenio_records_ui.signals import record_viewed
+
+from rero_ils.filter import format_date_filter
+from rero_ils.modules.documents.api import Document
+from rero_ils.modules.items.api import Item
+from rero_ils.modules.libraries.api import Library
+from rero_ils.modules.locations.api import Location
+
+from .api import CollectionsSearch
+from ..organisations.api import Organisation
+
+blueprint = Blueprint(
+    'collections',
+    __name__,
+    template_folder='templates',
+    static_folder='static',
+)
+
+
+def collection_view_method(pid, record, template=None, **kwargs):
+    """Display default view.
+
+    Sends record_viewed signal and renders template.
+    :param pid: PID object.
+    """
+    record_viewed.send(
+        current_app._get_current_object(), pid=pid, record=record)
+
+    # Get collections
+    search = CollectionsSearch().filter(
+        'term',
+        pid=pid.pid_value)
+
+    viewcode = kwargs['viewcode']
+    if (viewcode != current_app.config.get(
+        'RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'
+    )):
+        org_pid = Organisation.get_record_by_viewcode(viewcode)['pid']
+        search = search.filter(
+            'term', organisation__pid=org_pid
+        )
+
+    record['date'] = start_end_date(
+        record.get('start_date'), record.get('end_date'))
+
+    return render_template(
+        template,
+        record=record,
+        viewcode=viewcode
+    )
+
+
+def start_end_date(start_date, end_date):
+    """Format date."""
+    return '{start} - {end}'.format(start=format_date_filter(
+        start_date, date_format='short', time_format=None, locale='fr'),
+        end=format_date_filter(
+            end_date, date_format='short', time_format=None, locale='fr')
+    )
+
+
+@blueprint.app_template_filter()
+def get_teachers(record):
+    """Get teachers.
+
+    :param record: record
+    :return: list of teachers of the collection
+    """
+    teachers = []
+    for teacher in record.get('teachers'):
+        teachers.append(teacher.get('name'))
+    return ', '.join(teachers)
+
+
+@blueprint.app_template_filter()
+def get_items(record):
+    """Get items.
+
+    :param record: record
+    :return: list of items linked to collection
+    """
+    items = []
+    item_pids = []
+    for item in record.get('items'):
+        item_pids = [item['pid'] for item in
+                     record.replace_refs().get('items', [])]
+    for pid in item_pids:
+        items.append(Item.get_record_by_pid(pid))
+    return items
+
+
+@blueprint.app_template_filter()
```

Comment:
Is this function doesn't be placed into the `items/views.py` as it concerned an item, not a collection.